main.tf

```diff
@@ -46,9 +46,10 @@ locals {
   upgrade_command                              = "${local.gcloud} components update --quiet"
   additional_components_command                = "${path.module}/scripts/check_components.sh ${local.gcloud} ${local.components}"
   gcloud_auth_service_account_key_file_command = "${local.gcloud} auth activate-service-account --key-file ${var.service_account_key_file}"
+  activate_service_account                     = var.activate_service_account ? "${local.gcloud} auth activate-service-account --key-file ${local.tmp_credentials_path}" : "true"
   gcloud_auth_google_credentials_command       = <<-EOT
-    printf "%s" "$GOOGLE_CREDENTIALS" > ${local.tmp_credentials_path} &&
-    ${local.gcloud} auth activate-service-account --key-file ${local.tmp_credentials_path}
```

Comment:
I feel like this would bring breaking changes for other customers using this module

Reply:
`activate_service_account` defaults to true which means `local.activate_service_account` results to `${local.gcloud} auth activate-service-account --key-file ${local.tmp_credentials_path}` by default.

```
activate_service_account = var.activate_service_account ? "${local.gcloud} auth activate-service-account --key-file ${local.tmp_credentials_path}" : "true"
```

Reply:
it wont recreate anything because the default is exactly as before as explained above...

Reply:
Thanks @dogmatic69 - I did a quick pre/post test.